controllers/operands/networkAttachmentDefinitionHandler.go

```diff
@@ -14,18 +14,19 @@ import (
 	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 )
 
-type newNetworkAttachmentDefinitionFunc func(hc *hcov1beta1.HyperConverged) *netattdefv1.NetworkAttachmentDefinition
+type newNetworkAttachmentDefinitionFunc func(hc *hcov1beta1.HyperConverged, namespace string) *netattdefv1.NetworkAttachmentDefinition
```

Comment:
Maybe I miss something, but I don't understand why do we need this mess with this new function type. It can be dramatically simplified. The whole thing is just too complex.